src/components/main-view/main-view.jsx

```diff
@@ -140,8 +94,33 @@ export const MainView = () => {
         // });
 
         // userFavIDs = JSON.stringify(userData.Favorites);
-        setUserFavIDs(JSON.stringify(userData.Favorites));
+        setUserFavIDs(userData.Favorites);
+
+    }
 
+    async function logFav(id) {
+      /* 
+      Example of way to do this in memory.
```

Comment:
e.g. 

1. immediately store state updates in localStorage (session storage is even better IMO)
2. attempt to POST updates to API
3. if that succeeds - clear localStorage entries 
4. if that fails, either retry a set number of times, or fall back to localStorage 
5. Repeat